server/controllers/bookingControllers.js

```diff
@@ -1,184 +1,145 @@
+      tripid,
+    ]);
+    if (findtripid.rows.length <= 0) {
+      return response.response(res, 404, 'error', 'No trip found!', true);
+    }
+    if (findtripid) {
       // check if trip is activated*
-    if (findtripid.rows[0].status == 'CANCELED') {
-      return response.response(
-        res,
-        406,'error',
-  
-        'TRIP HAS BEEN CANCELED!!! PLEASE TRY ANOTHER DIFFERENT TRIP',
-        true
-      );
+      if (findtripid.rows[0].status === 'CANCELED') {
+        return response.response(res, 406, 'error', 'TRIP HAS BEEN CANCELED!!! PLEASE TRY ANOTHER DIFFERENT TRIP', true);
+      }
     }
 
+    const {
+      bus_license_number, origin, destination, fare, time, id, trip_date, seating_capacity,
+    } = findtripid.rows[0];
 
- }
- 
-  const buslicence = findtripid.rows[0].bus_license_number;
-  const origin  = findtripid.rows[0].origin;
-  const destination  = findtripid.rows[0].destination;
-  const fare = findtripid.rows[0].fare;
-  const triptime = findtripid.rows[0].time;
-  const tripidnumber = findtripid.rows[0].id;
-  const tripdate = findtripid.rows[0].trip_date;
-  const seating_capacity = findtripid.rows[0].seating_capacity;
-
-
-  // // ###check users  req.user.id
-  let userCheck = await client.query('SELECT * FROM users WHERE id=$1 ',[
-    req.user.id,
-  ]);
-  const userid = req.user.id;
-  // geting user info
-
-  const firstname = userCheck.rows[0].first_name;
-  const lastname = userCheck.rows[0].last_name;
-  const useremail = userCheck.rows[0].email;
-  const userphone = userCheck.rows[0].phonenumber;
- 
-
-  // check for booking and save
-  if (seating_capacity <= 0) {
-    return response.response(
-      res,
-      406,'error',
-      'SORRY!!! No seats left on the trip',
-      true
-    );
-    
-  }
 
+    // // ###check users  req.user.id
+    const userid = req.user.id;
+    const userCheck = await client.query('SELECT * FROM users WHERE id=$1 ', [
+      userid,
+    ]);
 
-  let book = await client.query('SELECT * FROM bookings WHERE trip_id=$1 AND user_id=$2',[
-    req.body.tripId,req.user.id,
-  ]);
+    // geting user info
+    const {
+      first_name, last_name, email, phonenumber,
+    } = userCheck.rows[0];
 
-  if (!book.rows[0]) {
-    // ##### inserting in table
+    // check for booking and save
 
-    let recordbooking = client.query('INSERT INTO bookings(booking_date, first_name,last_name, phonenumber, user_email, bus_license,origin,destination,trip_date,time,fare,trip_id,user_id)VALUES($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13)',[
-      moment().format(), firstname,lastname, userphone,useremail,buslicence,origin,destination,tripdate ,triptime,fare,tripidnumber,userid,
+    if (seating_capacity <= 0) { return response.response(res, 406, 'error', 'SORRY!!! No seats left on the trip', true); }
+    const book = await client.query('SELECT * FROM bookings WHERE trip_id=$1 AND user_id=$2', [
+      tripid, req.user.id,
```

Comment:
make use of the variable `userid` you have declared above instead of `req.user.id`